cpp/src/quantiles/quantiles.cu

```diff
@@ -92,12 +93,17 @@ namespace{ // anonymous
                              stream);
     }else{
       // create a clone of col_data if sort is required but sort_inplace is not allowed.
-      rmm::device_vector<ColType> dv(n);
-      thrust::copy_n(rmm::exec_policy(stream)->on(stream), col_data, n, dv.begin());
+      rmm::device_vector<ColType> dv(n-col_in->null_count);

```

Comment:
What about in the `if` branch above when `ctxt->flag_sort_inplace == true  and ctxt->flag_sorted == true`, will that support null values as well?

Reply:
It was not. I added the code to support null for sorted column. added gtest cases as well.